proj/src/main/webapp/script.js

```diff
@@ -27,12 +29,14 @@ if(btnLogin){
     });
 }
 
+var keyIdDict = {};
```

Comment:
I don't like this global `keyIdDict`. Is it possible to do this functionality without global variables like this. Like if this were an input to methods or was returned by methods.